### PR TITLE
Add PWM service

### PR DESCRIPTION
**Pull request by josiah-tesfu**
Implemented the PWM module for the RPC Javascript client. The implementation mirrors the PWM module in the Python RPC server. Included integration and unit tests.

This client module interfaces with the Python EdgePi RPC server through protobuf. It acts as a client for the Node-RED PWM module (not yet published).

Moved files inside the serviceTypes directory into a single file. This simplifies imports.

### CODE REVIEW

**Comment by josiah-tesfu** [comment]
Accidentally closed my last PR due to incorrect git usage. I resolved issues brought up there. Thanks!

**Review by flavius-t — APPROVED** [review]

File: src/services/PwmService/PwmService.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    const response: serverResponse = await this.rpcChannel.callMethod(
+      serviceReq,
+      requestType,
+      responseType
+    );
+
+    if (!response || response.error !== undefined) {
+      throw new Error(response ? response.error : "Response is null or undefined.");
+    }
+    return response;
```
Comment (jimmy121192) [inline src/services/PwmService/PwmService.ts:187]: These lines of code are a bit clunky (the condition in the IF block covers the condition of the exception). I think you can improve it. For example:

  if (response && !response.error) { //Return response if there is a response and NO error
      return response;
   } else {
    // Return any available error OR if the response is not available
      throw new Error(response && response.error || "Response is null or undefined.");
    }

File: src/services/PwmService/PwmService.ts [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  }
+
+  /**
+   * @async Calls the PWM close SDK method through an RPC request
+   */
+  async close(PWM: PWM): Promise<string> {
+    const response = await this.PWMRequest("close", PWM, SUCCESS_MSG);
+    return (response.content as SuccessMsg).content;
+  }
+
```
Comment (jimmy121192) [inline src/services/PwmService/PwmService.ts:75]: Will all the methods ( enable, disable, close, ...) be asynchronously executed in this function? Are we supposed to see `SuccessMsg` logs in the console?

File: tests/integration/PwmService.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    
+    test.each([
+        [{pwmNum: PWMPins.PWM1, frequency: 1000, dutyCycle: 1, polarity: Polarity.NORMAL}],
+        [{pwmNum: PWMPins.PWM2, frequency: 1000, dutyCycle: 1, polarity: Polarity.NORMAL}]
+    ])('Call setConfig and check configurations', async (args) => {
+        const response = await pwm.setConfig(args)
+        expect(response).toEqual('Successfully applied pwm configurations.')
+
+        const frequency = await pwm.getFrequency(args.pwmNum)
+        expect(typeof frequency).toBe('number')
```
Comment (jimmy121192) [inline tests/integration/PwmService.test.ts:29]: `typeof `is fine to use, but not for general type checking. I would recommend using Javascipt static methods to check:
For example: 
expect(Number.isFinite(dutyCycle)).toBe(true);

**Review by jimmy121192 — CHANGES_REQUESTED** [review]
I have added some comments. Also, pls add `.DS_Store` in the .gitignore file and install linter to check and format your JS code

**Review by maria-mz — DISMISSED** [review]

**Review by farzadpanahi — APPROVED** [review]

File: src/services/PwmService/PwmService.ts (hunk 2 above)
Comment (josiah-tesfu) [reply src/services/PwmService/PwmService.ts:75]: All methods are asynchronously executed. `SuccessMsg` is not logged in the console.

File: src/services/PwmService/PwmService.ts (hunk 1 above)
Comment (josiah-tesfu) [reply src/services/PwmService/PwmService.ts:187]: Thanks, that makes sense.

**Review by jimmy121192 — APPROVED** [review]

**Pull request merged**